core/src/data/value/literal.rs

```diff
@@ -204,6 +204,21 @@ impl Value {
 
                 Ok(Value::F64(v))
             }
+            (DataType::Decimal, Literal::Text(v)) => v
+                .parse::<Decimal>()
+                .map(Value::Decimal)
+                .map_err(|_| ValueError::LiteralCastFromTextToDecimalFailed(v.to_string()).into()),
+            (DataType::Decimal, Literal::Number(v)) => v
+                .to_string()
+                .parse::<Decimal>()
+                .map(Value::Decimal)
+                .map_err(|_| ValueError::FailedToParseDecimal(v.to_string()).into()),
```

Comment:
Let's unify the message `LiteralCastFromTextToDecimalFailed'.
Most of the errors I get in the `try_cast_from_literal` function seem to be `LiteralCastTo...`.

```suggestion
                .map_err(|_| ValueError::LiteralCastFromTextToDecimalFailed(v.to_string()).into()),
```